src/select/examples.js

```diff
@@ -11,159 +11,222 @@ LICENSE file in the root directory of this source tree.
-    // eslint-disable-next-line no-console
-    console.log('Selected option:' + option.id);
-  }
+const onChange = (...args) => {
+  // eslint-disable-next-line no-console
+  console.log('Selected option: ', ...args);
 };
 
-const onChangeMultiSelect = ({
-  event,
-  type,
-  option = emptyOption,
-  selectedOptions,
-}) => {
-  switch (type) {
-    case STATE_CHANGE_TYPE.select:
-      // eslint-disable-next-line no-console
-      console.log('Selected id:' + option.id);
-      break;
-    case STATE_CHANGE_TYPE.unselect:
-      // eslint-disable-next-line no-console
-      console.log('Unselected id:' + option.id);
-      break;
-  }
+const CustomValueWrapper = styled('span', {
+  display: 'inline-flex',
+  alignItems: 'center',
+});
+const CustomValueColor = styled('span', ({$color}) => ({
+  display: 'inline-block',
+  borderRadius: '50%',
+  width: '10px',
+  height: '10px',
+  backgroundColor: $color,
+  marginRight: '5px',
+}));
+function CustomOptionLabel({
+  option,
+  showRawColor = false,
+}: {
+  option: {id: string, color: string},
+  showColor?: boolean,
+}) {
+  return (
+    <CustomValueWrapper>
+      {option.color ? <CustomValueColor $color={option.color} /> : null}
+      {option.id}
+      {showRawColor && ` (${option.color})`}
+    </CustomValueWrapper>
+  );
+}
+
+const options = {
+  options: COLORS,
+  labelKey: 'id',
+  valueKey: 'color',
+  onChange: onChange,
+  placeholder: 'Choose a color',
 };
 
 export default {
-  [tests.SELECT]: () => {
-    return (
-      <StatefulSelect
-        options={COLORS}
-        getOptionLabel={option => option.id}
-        rows={10}
-        placeholder="Choose a color"
-        onChange={onChange}
-      />
-    );
-  },
-  [tests.SELECT_DISABLED_OPTIONS]: () => {
+  [tests.SELECT_NO_SEARCH]: () => {
     return (
       <StatefulSelect
+        {...options}
         options={[
           {
             id: 'red',
-            label: 'Red',
+            color: 'Red',
           },
           {
             id: 'orange',
-            label: 'Orange',
+            color: 'Orange',
             disabled: true,
           },
           {
             id: 'green',
-            label: 'Green',
+            color: 'Green',
           },
           {
             id: 'blue',
-            label: 'Blue',
+            color: 'Blue',
             disabled: true,
           },
         ]}
-        placeholder="Choose a color"
-        onChange={onChange}
+        searchable={false}
       />
     );
   },
-  [tests.MULTI_SELECT]: () => {
+  [tests.SELECT]: () => {
+    return <StatefulSelect {...options} maxDropdownHeight="300px" />;
+  },
+  [tests.SELECT_MULTI]: () => {
     return (
       <StatefulSelect
-        options={COLORS}
-        getOptionLabel={option => option.id}
-        multiple
+        {...options}
         initialState={{
-          selectedOptions: [COLORS[1], COLORS[3]],
+          value: [{...COLORS[0], clearableValue: false}, COLORS[1], COLORS[3]],
```

Comment:
Do we use it somewhere?
`clearableValue`

Reply:
Yes, check out the selected value functionality.